### DISCUSSION TITLE
Issue with Date Filtering: Prisma DateTime Format Mismatch with MySQL created_at Field

### DISCUSSION THREAD

**Discussion by Krrunal** [prisma/prisma#25296] [Q&A]
### Question

I am trying to filter records from the packages table where the created_at field is stored in the format YYYY-MM-DD HH:MM:SS.SSS (e.g., 2024-09-24 00:00:00.000). However, Prisma sends the date in ISO-8601 format with a T and Z (e.g., 2024-09-24T00:00:00.000Z), which causes issues with direct comparison against the created_at field in the database.

### Expected behavior (optional)

**Expected Behavior:**
Prisma should either:

Support the ability to compare dates using the YYYY-MM-DD HH:MM:SS.SSS format, or
Provide a way to adjust how Prisma formats date comparisons in queries to match the database's format.

### Information about Prisma Schema, Client Queries and Environment (optional)

**code Snippet**
```
prisma.package.findMany({
  where: {
    created_at: {
      lte: new Date("2022-01-30").toISOString(), // "2022-01-30T00:00:00.000Z"
      gte: new Date("2022-01-15").toISOString(), // "2022-01-15T00:00:00.000Z"
    },
    voys_status: {
      not: ""
    }
  },
  include: {
    store: true,
    route: true
  }
});

```
**Issue:**
Prisma automatically sends dates with the ISO-8601 format (2024-09-24T00:00:00.000Z) when I need it to match the MySQL format of 2024-09-24 00:00:00.000. This results in the query returning no results, even though records with matching created_at dates exist in the database.

**Attempted Solutions:**
Tried using the toISOString() method to convert dates into the ISO-8601 format, but this does not align with how the dates are stored in the database.
Attempted other date comparison techniques but was unable to find a way to match Prisma's generated date format with the database-stored format.

**Comment by ludralph** [comment]
Hi @Krrunal 👋 

Your database stores the date in a format that Prisma cannot directly handle. Unfortunately, there isn't a built-in feature to change how it formats date comparisons to match the database's format directly. Before sending it to Prisma, you must consider transforming the date format within your application logic.

For example:
```js
import { PrismaClient } from "@prisma/client";
import { format } from 'date-fns';

const startDate = format(new Date(2024, 9,4, 10, 24, 20), "yyyy-MM-dd'T'HH:mm:ss'z'")

const prisma = new PrismaClient();
async function main() {
  const res = await prisma.user.findMany({
    where:{
      createdAt:startDate
    }
  });
```

I'm using `date-fns` to format the dates in the `yyyy-MM-dd'T'HH:mm:ss'z'` format before passing them to the Prisma query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
